app/models/user.rb

```diff
@@ -8,12 +8,14 @@ class User < ApplicationRecord
 
   CST = 'Central Time (US & Canada)'
 
-  def current_week
-    if weeks.last && Time.now.in_time_zone(CST) < weeks.last.end
-      weeks.last
+  def find_week(week)
+    week_cst_time_zone = (Time.now + week).in_time_zone(CST)
+
+    if week_exists = weeks.find_by(end: week_cst_time_zone.end_of_week)
```

Comment:
Forgot to mention this in the initial review, but I find assignments like this (where you assign something and then have an if else) are hard to read so I usually break them up.

Also, another thing ot think about, this method is always returning a 'week' object, but this variable is called `week_exists` which makes it seem like its a true false return